src/main/java/com/betacampers/desafio_quality/repository/DistrictRepository.java

```diff
@@ -1,28 +1,111 @@
 package com.betacampers.desafio_quality.repository;
 
+import com.betacampers.desafio_quality.exception.DistrictNotFoundException;
 import com.betacampers.desafio_quality.model.District;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.springframework.core.io.ClassPathResource;
 import org.springframework.stereotype.Repository;
+import org.springframework.util.ResourceUtils;
 
-import java.math.BigDecimal;
-import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.Map;
-import java.util.stream.Collectors;
+import java.util.Properties;
 
 @Repository
 public class DistrictRepository implements IDistrictRepository {
 
     private static Map<Long, District> districts;
+    private String SCOPE;
 
     public DistrictRepository() {
-        districts = new ArrayList<District>() {{
-            add(new District(1, "Centro", new BigDecimal("10")));
-            add(new District(2, "Sossego", new BigDecimal("10")));
-            add(new District(3, "Passo D'areia", new BigDecimal("10")));
-        }}.stream().collect(Collectors.toMap(District::getDistrictId, x -> x));
+        Properties properties = new Properties();
+
+        try {
+            properties.load(new ClassPathResource("application.properties").getInputStream());
+            this.SCOPE = properties.getProperty("api.scope");
+            this.loadData();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
     }
 
     @Override
-    public District getById(long districtId) {
-        return districts.get(districtId);
+    public District getById(Long districtId) {
+        loadData();
+        District district = districts.get(districtId);
+        if (district != null) {
+            return district;
+        }
+        throw new DistrictNotFoundException(districtId);
+    }
+
+    @Override
+    public District save(District district) {
+        boolean districtNotExist = !exists(district);
+
+        if (district.getDistrictId() != null && districtNotExist) {
+            throw new DistrictNotFoundException(district.getDistrictId());
+        }
+
+        if (districtNotExist) {
+            Long greaterId = (districts.size() > 0) ? Collections.max(districts.keySet()) : 0L;
+            district.setDistrictId(greaterId + 1L);
+        }
+
+        districts.put(district.getDistrictId(), district);
+
+        this.saveData();
+
+        return district;
+    }
```

Comment:
Essa lógica pode ser feita no PropertyRepository também

Reply:
A lógica do id repliquei lá, mas sem fazer validação de existência, precisa disso para cadastro de imóvel?

Reply:
Para cadastro não, só se for atualizar o imóvel.